app/services/url_shortener.rb

```diff
@@ -0,0 +1,27 @@
+require 'net/http'
+require 'json'
+
+class UrlShortener
+  def initialize(http_utils: HttpUtils.new)
+    @http_utils = http_utils
+  end
+
+  def call(link)
+    begin
+      result = http_utils.post(
+        url: "https://api-ssl.bitly.com/v4/shorten",
+        token: A9n.bitly_access_token,
+        body: { "long_url" => link },
+      )
+      if http_utils.successful?(result)
```

Comment:
This obliges the client of `HttpUtils` to call twice on that. What if you return a `Result` as in https://github.com/HabitatMap/AirCasting/pull/137 ?

Reply:
Or you could do something like `post_or_default(default: long_link)` then http would return either the short or the long

Reply:
If `Result` had a method `or_value(default)` that gives back the value if `Success` or default if `Failure` then you could do
```
http_utils.post(...).or_value(long_link) 
```
like `Maybe.with_default`